newrelic/resource_newrelic_alert_policy_channel.go

```diff
@@ -83,56 +99,103 @@ func resourceNewRelicAlertPolicyChannelRead(d *schema.ResourceData, meta interfa
 	}
 
 	d.Set("policy_id", policyID)
-	d.Set("channel_id", channelID)
+
+	_, channelIDOk := d.GetOk("channel_id")
+	_, channelIDsOk := d.GetOk("channel_ids")
+
+	if channelIDOk && len(parsedChannelIDs) == 1 {
+		d.Set("channel_id", parsedChannelIDs[0])
+	}
+
+	if channelIDsOk && len(parsedChannelIDs) > 0 {
+		d.Set("channel_ids", parsedChannelIDs)
+	}
+
+	// Handle import (set `channel_ids` since this resource doesn't exist in state yet)
+	if !channelIDOk && !channelIDsOk {
+		d.Set("channel_ids", parsedChannelIDs)
```

Comment:
Hmm.  In the case where a new resource has only `channel_id` set, does this tell them about an inaccurate create during the plan since we're setting the `channel_ids` plural?

Reply:
Good call.

Reply:
After importing a subsequent `plan` will provide a deprecation warning message.